### ISSUE TITLE
Repacing optimalizer from 'Adam' to more powerful 'Adadelta'

### ISSUE THREAD

**Issue by tslapczynski** [GoogleCloudPlatform/asl-ml-immersion#404]
Now Adadelta is better optimalizer, so maybe it should be improved like this:

# Import the Adadelta optimizer
from transformers import Adadelta

# Define the optimizer
optimizer = Adadelta(model.parameters())

# Train the model
for epoch in range(num_epochs):
  model.train()

  for batch in train_loader:
    images = batch["images"]
    captions = batch["captions"]

    # Encode the images
    encoded_images = tokenizer(image=images, return_tensors="pt")

    # Generate the captions
    outputs = model(**encoded_images)

    # Calculate the loss
    loss = outputs.loss

    # Backpropagate the loss
    loss.backward()

    # Update the optimizer parameters
    optimizer.step()

# Evaluate the model
model.eval()

with torch.no_grad():
  for batch in val_loader:
    images = batch["images"]
    captions = batch["captions"]

    # Encode the images
    encoded_images = tokenizer(image=images, return_tensors="pt")

    # Generate the captions
    outputs = model(**encoded_images)

    # Calculate the CIDEr score
    cider_score = calculate_cider_score(outputs.captions, captions)

    # Print the CIDEr score
    print(cider_score)

**Comment by takumiohym** [comment]
Thanks for your suggestion.
But Adadelta is not regarded as a better optimizer as Adam. Rather, Adam tries to improve AdaDelta.
